runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPlugin.java

```diff
@@ -134,42 +134,55 @@ else if (worldType.contains(WorldType.LEAGUE))
 	@Subscribe
 	public void onInteractingChanged(InteractingChanged event)
 	{
-		if (event.getSource() != client.getLocalPlayer())
-		{
-			return;
-		}
+		Actor player = client.getLocalPlayer(), source = event.getSource(), target = event.getTarget();
 
-		Actor opponent = event.getTarget();
-
-		if (opponent == null)
+		if (source == player)
 		{
-			lastTime = Instant.now();
-			return;
+			if (target != null)
+			{
+				lastOpponent = target;
+				lastTime = null;
+			}
+			else if (lastOpponent != null
+					&& lastOpponent.getInteracting() != player)
+			{
+				lastTime = Instant.now();
+			}
 		}
+		else if (target == player
+				|| source == lastOpponent)
+		{
+			Actor opponent = player.getInteracting();
 
-		lastOpponent = opponent;
+			if (target != null && opponent == null)
+			{
+				lastOpponent = source;
+				lastTime = null;
+			}
+			else if (opponent != lastOpponent)
+			{
+				lastTime = Instant.now();
+			}
+		}
 	}
 
 	@Subscribe
 	public void onGameTick(GameTick gameTick)
 	{
-		if (lastOpponent != null
-			&& lastTime != null
-			&& client.getLocalPlayer().getInteracting() == null)
+		if (lastTime != null 
+			&& Duration.between(lastTime, Instant.now()).compareTo(WAIT) > 0)
 		{
-			if (Duration.between(lastTime, Instant.now()).compareTo(WAIT) > 0)
-			{
-				lastOpponent = null;
-			}
+			lastOpponent = null;
+			lastTime = null;
```

Comment:
Why is this nulled here now?

Reply:
Hello, so, in the current version of the plugin, it checks whether the player is interacting with an NPC on line 158 as part of determining whether to hide the health bar after WAIT time has passed or not.  Since my goal is to have the health bar visible while the player isn't technically interacting with the NPC but is still damaging it with recoil/cannon, I instead made lastTime null when it hasn't been set to a value in onInteractingChanged.  Otherwise, it would hide the health bar every tick after the initial WAIT time has passed.